firmware/controllers/trigger/trigger_central.cpp

```diff
@@ -257,6 +261,17 @@ void TriggerCentral::handleShaftSignal(trigger_event_e signal DECLARE_ENGINE_PAR
 
 	nowNt = getTimeNowNt();
 
+	// This code gathers some statistics on signals and compares accumulated periods to filter interference
+	if (boardConfiguration->useNoiselessTriggerDecoder) {
+		if (!noiseFilter(nowNt, signal PASS_ENGINE_PARAMETER_SUFFIX)) {
+			return;
+		}
+		// moved here from hwHandleShaftSignal()
+		if (!isUsefulSignal(signal, engineConfiguration)) {
+			return;
+		}
+	}
+
```

Comment:
Note that the default behavior wasn't changed (useNoiselessTriggerDecoder==false)